plugins/net_plugin/net_plugin.cpp

```diff
@@ -2489,7 +2488,18 @@ namespace eosio {
             boost::asio::bind_executor( strand,
               [conn = shared_from_this(), socket=socket]( boost::system::error_code ec, std::size_t bytes_transferred ) {
                // may have closed connection and cleared pending_message_buffer
-               if( !conn->socket_is_open() || socket != conn->socket ) return;
+               if (!conn->socket_is_open() && conn->socket_open) { // if socket_open then close not called
```

Comment:
This seems wrong. `conn->socket_is_open()` and `conn->socket_open` return the same `std::atomic` bool. So unless the strand gets interrupted between the two checks, the condition willnever be true.

Reply:
Thanks. I was reading that as `socket->is_open()`.